x-pack/plugins/observability_solution/observability_ai_assistant/server/functions/get_dataset_info/get_relevant_field_names.ts

```diff
@@ -36,9 +36,25 @@ export async function getRelevantFieldNames({
 }): Promise<{ fields: string[] }> {
   const dataViewsService = await dataViews.dataViewsServiceFactory(savedObjectsClient, esClient);
 
+  const hasAnyHitsResponse = await esClient.search({
+    index,
+    _source: false,
+    track_total_hits: 1,
+    terminate_after: 1,
+  });
+
+  const hitCount =
+    typeof hasAnyHitsResponse.hits.total === 'number'
+      ? hasAnyHitsResponse.hits.total
+      : hasAnyHitsResponse.hits.total?.value ?? 0;
```

Comment:
Remind me, why do we need to do this? Can't we force a consistent format?

Reply:
the type is ambiguous. figure it's safer/easier to write a ternary here instead of type casting